SingularityClient/src/main/java/com/hubspot/singularity/client/SingularityClient.java

```diff
@@ -114,10 +114,10 @@
   private static final String DELETE_DEPLOY_FORMAT = DEPLOYS_FORMAT + "/deploy/%s/request/%s";
 
   private static final String WEBHOOKS_FORMAT = "http://%s/%s/webhooks";
-  private static final String WEBHOOKS_DELETE_FORMAT = WEBHOOKS_FORMAT +"/%s";
-  private static final String WEBHOOKS_GET_QUEUED_DEPLOY_UPDATES_FORMAT = WEBHOOKS_FORMAT + "/deploy/%s";
-  private static final String WEBHOOKS_GET_QUEUED_REQUEST_UPDATES_FORMAT = WEBHOOKS_FORMAT + "/request/%s";
-  private static final String WEBHOOKS_GET_QUEUED_TASK_UPDATES_FORMAT = WEBHOOKS_FORMAT + "/task/%s";
+  private static final String WEBHOOKS_DELETE_FORMAT = WEBHOOKS_FORMAT +"?webhookId=%s";
+  private static final String WEBHOOKS_GET_QUEUED_DEPLOY_UPDATES_FORMAT = WEBHOOKS_FORMAT + "/deploy?webhookId=%s";
+  private static final String WEBHOOKS_GET_QUEUED_REQUEST_UPDATES_FORMAT = WEBHOOKS_FORMAT + "/request?webhookId=%s";
+  private static final String WEBHOOKS_GET_QUEUED_TASK_UPDATES_FORMAT = WEBHOOKS_FORMAT + "/task?webhookId=%s";
```

Comment:
It would be much cleaner to build the query params as part of the request object rather than tacking them on as part of the url. See `readSandBoxFile` in the client for an example of the query param builder, then you can use the `getCollectionWithParams` method inside `getQueued*` for the webhooks.